### PR TITLE
slack: map flags_mask and fix zerver_usermessage.

### PR DESCRIPTION
**Pull request by rheaparekh**

### CODE REVIEW

File: zulip/integrations/slack/slackdata2zulipdata.py [hunk 1]
```diff
@@ -339,21 +339,38 @@ def main(slack_zip_file: str) -> None:
     zerver_message = []
     zerver_usermessage = []
 
+    # check if a user has been mentioned in a message
+    def check_user_mention(text):
+        mentions = re.findall(r'(@(?:\*\*([^\*]+)\*\*|(\w+)))', text)
```
Comment (rht) [inline zulip/integrations/slack/slackdata2zulipdata.py:344]: The regex should be the same with the one at L230, shouldn't it?

Comment (rheaparekh) [reply zulip/integrations/slack/slackdata2zulipdata.py:344]: Here, we are taking the text from zerver_message which already has the sanitized text. So it'll be in the form `@**full_name**`. This is the regex for detecting this.

Comment (rht) [reply zulip/integrations/slack/slackdata2zulipdata.py:344]: IC

File: zulip/integrations/slack/slackdata2zulipdata.py [hunk 2]
```diff
@@ -339,21 +339,38 @@ def main(slack_zip_file: str) -> None:
...
-        for subscription in zerver_subscription:
-            if subscription['recipient'] == zulip_message['recipient']:
-                usermessage = dict(
-                    user_profile=subscription['user_profile'],
-                    id=usermessage_id,
-                    flags_mask=1,  # defaulting to 'read'
-                    message=zulip_message['id'])
-                usermessage_id += 1
-                zerver_usermessage.append(usermessage)
+    for message_index in range(len(zerver_message)):
```
Comment (rht) [inline zulip/integrations/slack/slackdata2zulipdata.py]: Why the extra for loop? I couldn't see `message_index` being use elsewhere.

Comment (rht) [reply zulip/integrations/slack/slackdata2zulipdata.py]: It looks to me as:
```python
for i in a:
    for j in a:
        ...
```

Comment (rheaparekh) [reply zulip/integrations/slack/slackdata2zulipdata.py]: So zerver_message is a list of list of dictionary. So we need two for loops to traverse for every single dictionary.

Comment (rheaparekh) [reply zulip/integrations/slack/slackdata2zulipdata.py]: That was the error I made initially. I assumed it to be a list of dicts. But for every channel, a list of dicts is being appended in the zerver_message list.

**Comment by rht** [comment]
Logic LGTM

**Pull request merged**